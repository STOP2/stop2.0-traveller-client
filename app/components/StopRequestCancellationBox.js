import React, { Component } from 'react'
import { View } from 'react-native'
import AwesomeButton from 'react-native-awesome-button'

import { DefaultText } from '../components/textComponents'

import styles from '../styles/stylesheet'
import colors from '../styles/colors'
import strings from '../resources/translations'

class StopRequestCancellationBox extends Component {
    render()
    {
        return (<View style={styles.stopRequestSentBackground}>
            <DefaultText style={styles.stopRequestSentText}>{this.props.supportsStopRequest ? strings.stopsent : strings.weWillNotifyYou} </DefaultText>
            <AwesomeButton labelStyle={styles.buttonLabel} states={{
                default: {
                    text: strings.cancel,
                    onPress: this.props.onPress,
                    backgroundColor: colors.HSLalarmRed
                }
            }} />
        </View>)
    }
}

StopRequestCancellationBox.propTypes = {
    onPress: React.PropTypes.func.isRequired,
    supportsStopRequest: React.PropTypes.bool.isRequired
}

export default StopRequestCancellationBox
