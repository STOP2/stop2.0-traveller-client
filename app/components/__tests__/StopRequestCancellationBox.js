import 'react-native'
import React from 'react'
import StopRequestCancellationBox from '../StopRequestCancellationBox'

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer'


jest.mock('react-native-localization')

describe('StopRequestCancellationBox component', () =>
{
    it('renders correctly', () =>
    {
        let tree = renderer.create(<StopRequestCancellationBox supportsStopRequest={true} onPress={() => {}} />).toJSON()

        expect(tree).toMatchSnapshot()
    })
})
