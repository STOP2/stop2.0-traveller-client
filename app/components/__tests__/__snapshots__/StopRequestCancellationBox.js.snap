exports[`StopRequestCancellationBox component renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#BEE4F8",
      "height": undefined,
      "padding": 10,
      "width": undefined,
    }
  }>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontFamily": "gotham-rounded-book",
        },
        Object {
          "fontSize": 20,
          "fontWeight": "bold",
          "marginBottom": 10,
        },
      ]
    }>
    Stop request sent
     
  </Text>
  <View
    accessibilityComponentType="button"
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "flex": 1,
        "opacity": 1,
      }
    }
    testID={undefined}>
    <View
      style={
        Object {
          "backgroundColor": "rgba(220, 4, 81, 1)",
          "borderRadius": 20,
          "flex": 1,
          "height": 40,
        }
      }>
      <View
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "flex": 1,
            "flexDirection": "row",
          }
        }>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "white",
              "fontFamily": "gotham-rounded-medium",
              "fontSize": 20,
            }
          }>
          Cancel
        </Text>
      </View>
    </View>
  </View>
</View>
`;
